src/Moryx.AbstractionLayer/Resources/Initializer/ResourceInitializerBase.cs

```diff
@@ -56,6 +56,42 @@ void IPlugin.Stop()
         }
 
         /// <inheritdoc />
-        public abstract IReadOnlyList<Resource> Execute(IResourceGraph graph);
+        public abstract Task<ResourceInitializerResult> Execute(IResourceGraph graph, object parameters);
+
+        /// <summary>
+        /// Creates an <see cref="ResourceInitializerResult"/> within a completed task
+        /// </summary>
+        /// <param name="initializedResources">Initialized root resources</param>
+        protected static Task<ResourceInitializerResult> InitializedAsync(IReadOnlyList<Resource> initializedResources) =>
+            InitializedAsync(initializedResources, false);
+
+        /// <summary>
+        /// Creates an <see cref="ResourceInitializerResult"/> within a completed task
+        /// </summary>
+        /// <param name="initializedResources">Initialized root resources</param>
+        /// <param name="saved">If true, the roots will not be saved again after result</param>
```

Comment:
Sorry but I con't understand from the comment, what does the `saved` paramter do?